### PR TITLE
Update 'Find' Command

### PR DESCRIPTION
**Pull request by choonyongchan**
The previous 'Find' command can only support multiple 'Name' keywords.

We want Socius to support all properties (e.g. 'Social', 'Tutorial Group', etc)

This commit implements an updated 'Find' command that supports multiple keywords of any properties. The format follows 'Add' and 'Edit'. A 'MultipleParsers' class was implemented to easily track how many of what Predicates are in use in a 'Find' command.

### CODE REVIEW

File: src/main/java/seedu/address/model/person/EmailContainsKeywordsPredicate.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
```
Comment (hsiaotingluv) [inline src/main/java/seedu/address/model/person/EmailContainsKeywordsPredicate.java:9]: Is it @code Phone or @code Email?

File: src/main/java/seedu/address/model/person/NationalityContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
```
Comment (hsiaotingluv) [inline src/main/java/seedu/address/model/person/NationalityContainsKeywordsPredicate.java:10]: minor change from {@code Phone} to {@code Nationality}

File: src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
```
Comment (hsiaotingluv) [inline src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java:10]: same as above :))

File: src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+
+
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches any of the keywords given.
```
[the remark this answers is not in the document]
Comment (choonyongchan) [reply src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java:11]: Thanks for catching these documentation errors. I've fixed it and will update the PR. 
Only for this class, the documentation seems ok.

**Comment by choonyongchan** [comment]
Thanks Hsiao Ting, I've fixed the documentation errors, together with introducing new test cases for 'FindCommand' and the new Parsers.
(ps: Thanks for reviewing this mess ㅠㅠ)

**Review by hsiaotingluv — APPROVED** [review]

**Pull request merged**